### DISCUSSION TITLE
usbAudio ?

### DISCUSSION THREAD

**Discussion by OxF8** [tobozo/ESP32-USB-Soft-Host#33] [Q&A]
Hello Dear, is it possible  USB Audio Class 1 standard? And to connect ESP32 as external usb sound card :)

**Comment by tobozo** [comment]
Hi,

indeed USB 1.0 High Speed can support PCM digital audio up to 24 bits/96kHz, but ESP32-USB-Soft-Host is limited to Low Speed devices, and does not implement the audio class. 

For a USB audio support, you'll be better with an ESP32-S2 or S3 using an USB library that implements audio at full speed:

https://github.com/pschatzmann/arduino-audio-tools/tree/main/src

[edit] 
I'm curious about the device though, most USB audio dongles I've tried are based on USB 1.1 so they aren't even seen by ESP32-USB-Soft-Host, but some USB PCM microphones could match the requirements.

![image](https://user-images.githubusercontent.com/1893754/233608676-54e1ebed-b9bd-42e1-9fbe-4ea359a7df9b.png)

If your device is similar, then chances are it could be detected by a modified version of the library, either way I'll need more information on said audio device.

**Reply by OxF8** [reply]
In fact I want to use ESP32, connected to USB laptop, as external soundcard. And I will wait for S2. Didn't test anything with soft usb host.

**Reply by tobozo** [reply]
maybe the S3 is a better pick then, it has 2x I2s interface and 2x cores whereas the S2 only has one of each

a device like [sonatino](https://sonatino.com) would be perfect for that

**Reply by OxF8** [reply]
Thank You, I will wait for S3 in such a case. To be honest - I didn't check hardware :) And Sonatino looks great, but I'm in doubts, I will find it.
